src/zino/tasks/bfdtask.py

```diff
@@ -0,0 +1,84 @@
+import logging
+from typing import Dict
+
+from zino.scheduler import get_scheduler
+from zino.snmp import SNMP, SparseWalkResponse
+from zino.statemodels import (
+    BFDEvent,
+    BFDSessState,
+    BFDState,
+    DeviceState,
+    EventState,
+    Port,
+)
+from zino.tasks.task import Task
+
+_log = logging.getLogger(__name__)
+
+BFDStates = Dict[str, BFDState]
+
+
+class BFDTask(Task):
+    JUNIPER_BFD_COLUMNS = [
+        ("BFD-STD-MIB", "bfdSessState"),
+        ("JUNIPER-BFD-MIB", "jnxBfdSessIntfName"),  # This should match IfDescr from the IF-MIB
+    ]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._scheduler = get_scheduler()
+        self._snmp = SNMP(self.device)
+
+    async def run(self):
+        if self._device_state.is_juniper:
+            polled_state = await self._poll_juniper()
+            self._update_state_for_all_ports(polled_state)
+
+    def _update_state_for_all_ports(self, polled_state: BFDStates):
+        for port in self._device_state.ports.values():
+            new_state = polled_state.get(port.ifdescr, BFDState(session_state=BFDSessState.NOSESSION))
+            self._update_state(port, new_state)
```

Comment:
Just a question: Does the original code actually keep a state for ports that do not have any BFD information?

Reply:
Ill have to look into that. I've mostly disregarded the original implementation since it was known to be based on incorrect assumptions